aesara/tensor/basic_opt.py

```diff
@@ -1645,6 +1645,10 @@ def dimshuffled_alloc(i):
             new_i.append(i)
```

Comment:
Problem is that we hit this `else` bucket which should only apply to the inputs that are not relevant for the rewrite...